src/reducers/select.js

```diff
@@ -0,0 +1,190 @@
+/**
+*
+* @licstart  The following is the entire license notice for the JavaScript code in this file.
+*
+* Merge MARC records
+*
+* Copyright (C) 2015-2019 University Of Helsinki (The National Library Of Finland)
+*
+* This file is part of marc-record-merge-js
+
+* marc-record-merge-js program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* marc-record-merge-js is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @licend  The above is the entire license notice
+* for the JavaScript code in this file.
+*
+*/
+import createDebugLogger from 'debug';
+import {normalizeSync} from 'normalize-diacritics';
+
+// The equalityFunction can be either strictEquality or subsetEquality
+// Returns true or false
+export function strictEquality(subfieldA, subfieldB) {
+  return subfieldA.code === subfieldB.code &&
+  subfieldA.value === subfieldB.value;
+}
+
+export function subsetEquality(subfieldA, subfieldB) {
+  // eslint error message: Unexpected mix of '&&' and '||'
+  /* eslint-disable */
+  return subfieldA.code === subfieldB.code &&
+  subfieldA.value.indexOf(subfieldB.value) !== -1 || subfieldB.value.indexOf(subfieldA.value) !== -1;
+  /* eslint-enable */
```

Comment:
No need to disable ESLint. See the rule [description](https://eslint.org/docs/rules/no-mixed-operators#disallow-mixes-of-different-operators-no-mixed-operators) (`Enclosing complex expressions by parentheses clarifies the developer's intention, which makes the code more readable. This rule warns when different operators are used consecutively without parentheses in an expression.`).

So wrap in parentheses as described above. Also, if rules are to be disabled, they must disabled as follows: `// eslint-disable-line <rule>`